### ISSUE TITLE
Session.CreateSession(...) fails on response unmarshaling

### ISSUE THREAD

**Issue by belyshevdenis** [okta/okta-sdk-golang#95]
[Session.CreateSession method](https://github.com/okta/okta-sdk-golang/blob/master/okta/session.go#L69) fails with error: `json: cannot unmarshal string into Go struct field Session.amr of type okta.SessionAuthenticationMethod`. Example of body:
```json
{ 
   "id":"102qtF5AqKmQ4yK34D_EJ1YDg",
   "userId":"00unkik37pLSLTUmV0h7",
   "login":"marty61mcfly@mail.ca",
   "createdAt":"2019-09-26T11:02:59.000Z",
   "expiresAt":"2019-09-26T13:02:59.000Z",
   "status":"ACTIVE",
   "lastPasswordVerification":"2019-09-26T11:02:59.000Z",
   "lastFactorVerification":null,
   "amr":[ 
      "pwd"
   ],
   "idp":{ 
      "id":"00oemxjehmcnwkqOT0h7",
      "type":"OKTA"
   },
   "mfaActive":false,
   "_links":{ 
      "self":{ 
         "href":"https://dev-9xxxxx.oktapreview.com/api/v1/sessions/me",
         "hints":{ 
            "allow":[ 
               "GET",
               "DELETE"
            ]
         }
      },
      "refresh":{ 
         "href":"https://dev-9xxxxx.oktapreview.com/api/v1/sessions/me/lifecycle/refresh",
         "hints":{ 
            "allow":[ 
               "POST"
            ]
         }
      },
      "user":{ 
         "name":"Marty McFly",
         "href":"https://dev-9xxxxx.oktapreview.com/api/v1/users/me",
         "hints":{ 
            "allow":[ 
               "GET"
            ]
         }
      }
   }
}
```
Type `okta.SessionAuthenticationMethod` corresponds to `arm` field in body example.

**Comment by belyshevdenis** [comment]
It is a generator bug:
The `amr` property declared as enum as strings in spec: https://github.com/okta/okta-sdk-golang/blob/master/openapi/spec.json#L6017
But the generated type is empty struct: https://github.com/okta/okta-sdk-golang/blob/master/okta/sessionAuthenticationMethod.go#L23

**Comment by bretterer** [comment]
I have submitted an update that will be released in v2.0.0-rc.5 to make all the objects that are enums in the spec as `type string` instead.  This will solve the issues we have been having around this.  

Resolved in 01d8997
